fabric-enchanting-api-v1/src/main/java/net/fabricmc/fabric/mixin/enchanting/MixinEnchantingTableContainer.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.enchanting;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.ModifyVariable;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.container.EnchantingTableContainer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.World;
+
+import net.fabricmc.fabric.api.enchanting.v1.EnchantingPowerProvider;
+
+@Mixin(EnchantingTableContainer.class)
+public abstract class MixinEnchantingTableContainer {
+	@Unique
+	private World world;
+	@Unique
+	private BlockPos blockPos;
+
+	@SuppressWarnings("UnresolvedMixinReference")
+	@Inject(method = "method_17411(Lnet/minecraft/item/ItemStack;Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)V", at = @At("HEAD"))
+	private void onEnchantmentCalculation(ItemStack itemStack, World world, BlockPos blockPos, CallbackInfo callbackInfo) {
+		this.world = world;
+		this.blockPos = blockPos;
+	}
+
+	@SuppressWarnings("UnresolvedMixinReference")
+	@ModifyVariable(method = "method_17411(Lnet/minecraft/item/ItemStack;Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;)V", at = @At(value = "INVOKE", target = "Ljava/util/Random;setSeed(J)V", shift = At.Shift.BEFORE), ordinal = 0)
+	private int changeEnchantingPower(int power) {
```

Comment:
You may notice the signature of the synthetic method you are using modify variable in contains an ItemStack world and BlockPos.

You can change the method signature to the following since mixin allows you to use parameters from the method that a specific targeting uses.
```java
private int changeEnchantingPower(int power, ItemStack stack, World world, BlockPos pos) {
```

In that case then you will need to pass this world and other details to the method below this one in order to remove the need to keep the fields.

Reply:
whoops that stems from the time where I didn't know yet that one can do this xD